python/grass/jupyter/testsuite/display_test.py

```diff
@@ -0,0 +1,125 @@
+        This is executed after each test run.
+        """
+        for f in self.files:
+            Path(f).unlink(missing_ok=True)
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_defaults(self):
+        """Test that GrassRenderer can create a map with default settings."""
+        # Create a map with default inputs
+        grass_renderer = gj.GrassRenderer()
+        # Adding vectors and rasters to the map
+        grass_renderer.run("d.rast", map="elevation")
+        grass_renderer.run("d.vect", map="roadsmajor")
+        # Assert image exists
+        self.assertFileExists(grass_renderer._filename)
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_filename(self):
+        """Test that GrassRenderer creates maps with unique filenames."""
+        # Create map with unique filename
+        grass_renderer = gj.GrassRenderer(filename="test_filename.png")
+        # Add files to self for cleanup later
+        self.files.append("test_filename.png")
+        self.files.append("test_filename.grass_vector_legend")
+        # Add a vector and a raster to the map
+        grass_renderer.run("d.rast", map="elevation")
+        grass_renderer.run("d.vect", map="roadsmajor")
+        # Assert image exists
+        self.assertFileExists("test_filename.png")
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_hw(self):
+        """Test that GrassRenderer creates maps with unique height and widths."""
+        # Create map with height and width parameters
+        grass_renderer = gj.GrassRenderer(width=400, height=400)
+        # Add just a vector (for variety here)
+        grass_renderer.run("d.vect", map="roadsmajor")
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_env(self):
+        """Test that we can hand an environment to GrassRenderer."""
+        # Create map with environment parameter
+        grass_renderer = gj.GrassRenderer(env=os.environ.copy())
+        # Add just a raster (again for variety)
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_text(self):
+        """Test that we can set a unique text_size in GrassRenderer."""
+        # Create map with unique text_size parameter
+        grass_renderer = gj.GrassRenderer(text_size=10)
+        # Add a vector and a raster
+        grass_renderer.run("d.vect", map="roadsmajor")
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(can_import_folium() is False, "Cannot import folium")
+    def test_shortcut(self):
+        """Test that we can use display shortcuts with __getattr__."""
+        # Create map
+        grass_renderer = gj.GrassRenderer()
+        # Add raster to map with shortcut
+        grass_renderer.d_rast(map="elevation")
```

Comment:
Testing more modules than just d.rast would be nice, at least d.vect.

You can also test that it throws attr error when the module name is wrong. See [assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) and use it as a context manager (with `with`). You could also test with `assertRaisesRegex` and confirm that module name is in the message and things like that.

Reply:
I added d.vect and another test for the attr error.